File: benchmark/java/Strings.java
// Strings.java - String Operations Benchmark for Java
// Tests string concatenation and manipulation performance

public class Strings {
    public static int countOccurrences(String str, String substr) {
        int count = 0;
        int pos = 0;
        while ((pos = str.indexOf(substr, pos)) != -1) {
            count++;
            pos++;
        }
        return count;
    }

    public static void main(String[] args) {
        int count = 100000;
        String hello = "Hello";

        long start = System.currentTimeMillis();

        // Build string using StringBuilder
        StringBuilder builder = new StringBuilder(count * hello.length());
        for (int i = 0; i < count; i++) {
            builder.append(hello);
        }
        String result = builder.toString();

        // Count occurrences of 'llo'
        int occurrences = countOccurrences(result, "llo");

        long end = System.currentTimeMillis();

        System.out.println("String length: " + result.length());
        System.out.println("Occurrences of 'llo': " + occurrences);
        System.out.println("String time: " + (end - start) + "ms");
    }
}

File: benchmark/go/strings.go
// strings.go - String Operations Benchmark for Go
// Tests string concatenation and manipulation performance
package main

import (
	"fmt"
	"strings"
	"time"
)

func countOccurrences(s, substr string) int {
	count := 0
	pos := 0
	for {
		idx := strings.Index(s[pos:], substr)
		if idx == -1 {
			break
		}
		count++
		pos += idx + 1
	}
	return count
}

func main() {
	count := 100000
	hello := "Hello"

	start := time.Now()

	// Build string using strings.Builder for efficiency
	var builder strings.Builder
	builder.Grow(count * len(hello))
	for i := 0; i < count; i++ {
		builder.WriteString(hello)
	}
	result := builder.String()

	// Count occurrences of 'llo'
	occurrences := countOccurrences(result, "llo")

	elapsed := time.Since(start).Milliseconds()

	fmt.Printf("String length: %d\n", len(result))
	fmt.Printf("Occurrences of 'llo': %d\n", occurrences)
	fmt.Printf("String time: %dms\n", elapsed)
}
